kubetest2-gke/deployer/up.go

```diff
@@ -99,7 +99,10 @@ func (d *Deployer) CreateClusters() error {
 
 		r := retryCount
 		if err := eg.Wait(); err != nil {
-			if d.isRetryableError(err) {
+			// If the error is retryable and it is not the last region/zone that
+			// can be retried, perform cleanups in the background and retry
+			// cluster creation in the next available region/zone.
+			if d.isRetryableError(err) && r != totalTryCount-1 {
 				go func() {
```

Comment:
if r == totalTryCount-1 we're never going to cleanup, right?

go prefers not to indent the common cases

Reply:
if `r == totalTryCount-1` we will not clean up the clusters asynchronously. Instead, we will clean them up in the `Down` function if `--down` flag is provided. This conforms with the behavior when no retry is needed.